### PR TITLE
Create sandbox k8s-infra-porche project

### PR DESCRIPTION
**Pull request by justinsb**
A copy-and-replace of k8s-infra-oci-proxy.

### CODE REVIEW

**Comment by justinsb** [comment]
/hold cancel - I think we have all the prereqs in place for this now

Note that this maybe should be called `k8s-infra-porche-staging`, but I'm copying `k8s-infra-oci-proxy` which has `k8s-infra-oci-proxy` and `k8s-infra-oci-proxy-prod`.  But if we want to add the `-staging` suffix here, now is the time I think :-)

**Comment by ameukam** [comment]
> /hold cancel - I think we have all the prereqs in place for this now
> 
> Note that this maybe should be called `k8s-infra-porche-staging`, but I'm copying `k8s-infra-oci-proxy` which has `k8s-infra-oci-proxy` and `k8s-infra-oci-proxy-prod`. But if we want to add the `-staging` suffix here, now is the time I think :-)

+1 for the `-staging` suffix. At the time we didn't to plan to have more than one env for `archeio`.

File: infra/gcp/terraform/k8s-infra-porche/provider.tf [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+/*
+This file defines:
+- Required provider versions
+- Storage backend details
+*/
+
+terraform {
+  backend "gcs" {
+    bucket = "k8s-infra-tf-porche"
```
Comment (ameukam) [inline infra/gcp/terraform/k8s-infra-porche/provider.tf:25]: You'll need to set this bucket in the `kubernetes-public` project.

Comment (justinsb) [reply infra/gcp/terraform/k8s-infra-porche/provider.tf]: Thanks - added a commit that (I think) does this!

**Comment by justinsb** [comment]
Updated with the `-staging` suffix.  I don't think this is blocked on the buckets at this point.  It's not going to save us any money until we can redirect to S3, but at least it brings us one step closer to saving money :-)

**Comment by ameukam** [comment]
/approve
/lgtm
(currently traveling and away from upstream). we'll actuate the script this week.

**Comment by justinsb** [comment]
Awesome thank you @dims and @ameukam .  I applied a hold and failed to cancel it, but I wasn't sure if we should keep the hold until this can be applied.  From my perspective, I don't think I need to hold it any more.

**Comment by justinsb** [comment]
Can we apply this?  We now have some test AWS buckets and I've populated a few files and the redirector (porche) seems to be working pretty nicely!

**Comment by ameukam** [comment]
Partial application of the HCL code:

```bash
tf apply -auto-approve -target='google_project.project' -target='google_project_service.project'
....
Apply complete! Resources: 10 added, 0 changed, 0 destroyed.
```

```bash
gcloud projects describe k8s-infra-porche-sandbox
createTime: '2023-02-01T22:48:04.054Z'
lifecycleState: ACTIVE
name: k8s-infra-porche-sandbox
parent:
  id: '758905017065'
  type: organization
projectId: k8s-infra-porche-sandbox
projectNumber: '795576063597'
```

```bash
 gcloud services list --project k8s-infra-porche-sandbox
NAME                              TITLE
compute.googleapis.com            Compute Engine API
containerregistry.googleapis.com  Container Registry API
logging.googleapis.com            Cloud Logging API
monitoring.googleapis.com         Cloud Monitoring API
oslogin.googleapis.com            Cloud OS Login API
pubsub.googleapis.com             Cloud Pub/Sub API
run.googleapis.com                Cloud Run Admin API
storage-api.googleapis.com        Google Cloud Storage JSON API
storage-component.googleapis.com  Cloud Storage
```

@justinsb You should be able to create the rest of the resources.

**Comment by ameukam** [comment]
Ensuring the google group is admin:

```bash
tf apply -auto-approve -target='google_project_iam_member.k8s_infra_porche_admins'


Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # google_project_iam_member.k8s_infra_porche_admins will be created
  + resource "google_project_iam_member" "k8s_infra_porche_admins" {
      + etag    = (known after apply)
      + id      = (known after apply)
      + member  = "group:k8s-infra-porche-admins@kubernetes.io"
      + project = "projects/k8s-infra-porche-sandbox"
      + role    = "roles/owner"
    }
...
Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

**Pull request merged**